openid-4-verifiable-presentations-1_0.md

```diff
@@ -1837,24 +1852,56 @@ Considerable discourse regarding unlinkability in salted-hash based selective di
 unlinkability properties is the use of batch issuance, which is supported in [@?OpenID4VCI], with individual credentials
 being presented only once.
 
-## Authorization Requests with Request URI
+Note that the nature of the issuer of the credential may leak unnecessary information such as nationality or state of residence.
+
+### Strictly Necessary {#strictly_necessary}
+
+* Verifiers SHOULD design queries that request only the minimal set of claims and credentials needed to fulfill the specified purposes. DCQL helps facilitate this. 
+
+### No Fingerprinting {#no_fingerprinting}
+
+Verifier SHOULD NOT attempt to fingerprint the wallets to track the user's visits.
+
+
+## Data Minimization {#data_minimization}
+
+### Wallet to verifier communication {#wallet_to_verifier_communication}
 
-If the Wallet is acting within a trust framework that allows the Wallet to determine whether a 'request_uri' belongs to a certain 'client_id', the Wallet is RECOMMENDED to validate the Verifier's authenticity and authorization given by 'client_id' and that the 'request_uri' corresponds to this Verifier. If the link cannot be established in those cases, the Wallet SHOULD refuse the request or ask the End-User for advise.
+*   **HTTP Headers:** Wallets SHOULD only send the minimal amount of information possible, and in particular, without any HTTP headers identifying the software used for the request (e.g., HTTP libraries or their versions) when retrieving a `request_uri` or sending to `response_uri` to reduce the risk of fingerprinting and End-User tracking. 
+    
+*   **No unauthorized PII:** Wallets MUST NOT include any personally identifiable information (PII) in HTTP requests to Verifiers unless explicitly required for the flow and authorized by the user. 
+    
+*   **No unauthorized request to Request URI:** If no End-User interaction is required before sending the request to Request URI, it is easy for a malicious verifier to obtain the wallet capabilities from all visitors of a website on a large scale and in an automated fashion. Even without personally identifiable information (PII) this can reveal some information about End-Users, like their nationality (e.g., a Wallet with special capabilities only used in one EU member state). Mandatory End-User interaction before sending the request, like clicking a button, unlocking the wallet or even just showing a screen of the app, can make this less attractive/likely to being exploited.
```

Comment:
```suggestion
*   **No unauthorized request to Request URI:** If no End-User interaction is required before sending the request to Request URI, it is easy for a malicious Verifier to obtain the Wallet capabilities from all visitors of a website on a large scale and in an automated fashion. Even without personally identifiable information (PII) this can reveal some information about End-Users, like their nationality (e.g., a Wallet with special capabilities only used in one EU member state). Mandatory End-User interaction before sending the request, like clicking a button, unlocking the Wallet or even just showing a screen of the app, can make this less attractive/likely to being exploited.
```